fmt/stm.c

```diff
@@ -167,21 +173,27 @@ int fmt_stm_load_song(song_t *song, slurp_t *fp, unsigned int lflags)
 	slurp_read(fp, tmp, 4);
 
 	if (!(
-		// this byte is guaranteed to be 0x1a, always...
-		tmp[0] == 0x1a
+		// this byte is *usually* guaranteed to be 0x1a,
+		// however putup10.stm and putup11.stm are outliers
+		// for some reason?...
+		(tmp[0] == 0x1a || tmp[0] == 0x02)
 		// from the doc:
 		//      1 - song (contains no samples)
 		//      2 - module (contains samples)
 		// I'm not going to care about "songs".
 		&& tmp[1] == 2
-		// and check the file tag -- but god knows why, it's case insensitive
-		&& (strncasecmp(id, "!Scream!", 8) == 0 || strncasecmp(id, "BMOD2STM", 8) == 0)
+		// do version 1 STM's even exist?...
+		&& tmp[2] == 2
 	)) {
 		return LOAD_UNSUPPORTED;
 	}
+	// check the file tag for printable ASCII
+	for (n = 0; n < 8; n++)
+		if (id[n] < ' ' || id[n] > '~')
```

Comment:
Somewhat personal preference, but this ought to use hex literals instead of char literals because char literals can virtually be any encoding (but are almost always ASCII)